packages/react/src/components/tag/Tag.tsx

```diff
@@ -25,60 +27,60 @@ export enum TagSize {
   Large = 'large',
 }
 
-export type TagProps = {
-  /**
-   * The label for the tag
-   */
-  children: string;
-  /**
-   * Additional class names to apply to the tag
-   */
-  className?: string;
-  /**
-   * Url to go to after Link variant is clicked
-   */
-  href?: string;
-  /**
-   * Link's _target -attribute
-   */
-  target?: React.AnchorHTMLAttributes<HTMLAnchorElement>['target'];
-  /*
-  /**
-   * Element placed on the starting side of the label
-   */
-  iconStart?: React.ReactNode;
-  /**
-   * Element placed on the ending side of the label
-   */
-  iconEnd?: React.ReactNode;
-  /**
-   * Should Tag span to multiple lines
-   * @default false
-   */
-  multiline?: boolean;
-  /**
-   * Callback function fired when the tag is clicked. If set, the tag will be clickable.
-   */
-  onClick?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
-  /**
-   * Callback function fired when the tag is clicked. If set, a delete button will be shown.
-   */
-  onDelete?: (event: React.MouseEvent<HTMLDivElement, MouseEvent> | React.KeyboardEvent<HTMLDivElement>) => void;
-  /**
-   * Size variant for the Tag.
-   * @default TagSize.Small
-   */
-  size?: TagSize;
-  /**
-   * Custom theme styles
-   * Will contain more properties in the next major release.
-   */
-  theme?: TagTheme;
-  /**
-   * Ref is set to the main element
-   */
-  ref?: React.Ref<HTMLDivElement> | React.Ref<HTMLAnchorElement>;
-};
+export type TagProps = CommonProps &
+  IntersectType<React.AnchorHTMLAttributes<HTMLAnchorElement>, React.HTMLAttributes<HTMLDivElement>> & {
```

Comment:
I'd bet the Div-version is more commonly used. We could also force the devs to give the type when Anchor-version is used:

```
export type TagProps<T = HTMLDivElement|> = CommonProps & React.HTMLAttributes<T> & ....
```

Usage:
```
const divProps: TagProps = {}
const anchorProps: TagProps<HTMLAnchorElement> = {}
```